### PR TITLE
Shading Language documentation proofread, fixed errors

### PR DESCRIPTION
**Pull request by eluadyl**
checked 'shading_language.rst' for typos and grammar errors, made the necessary changes being careful not to alter any meanings.

### CODE REVIEW

File: learning/features/shading/shading_language.rst [hunk 1]
```diff
@@ -174,7 +174,7 @@ It is possible to add precision modifiers to datatypes, use them for uniforms, v
 Using lower precision for some operations can speed up the math involved (at the cost of, of course, less precision).
 This is rarely needed in the vertex shader (where full precision is needed most of the time), but often needed in the fragment one.
 
-Keep in mind that some architectures (mainly mobile) benefit a lot on this, but are also restricted (conversion between precisions has a cost).
+Keep in mind that some architectures (mainly mobile) benefit a lot from this, but are also restricted (conversion between precisions has a cost).
 Please read the relevant documentation on the target architecture to find out more. In all honesty though, mobile drivers are really buggy
 so just stay out of trouble and make simple shaders without specifying precission unless you *really* need to.
```
Comment (Noshyaar) [inline learning/features/shading/shading_language.rst]: `precission`

File: learning/features/shading/shading_language.rst [hunk 2]
```diff
@@ -763,7 +763,7 @@ Fragment Built-Ins
 +----------------------------------+--------------------------------------------------------------------------------------------------+
 | out vec3 **NORMALMAP**           | Output this if reading normal from a texture instead of NORMAL.                                  |
 +----------------------------------+--------------------------------------------------------------------------------------------------+
-| out float **NORMALMAP_DEPTH**    | Depth from variable avobe. Defaults to 1.0.                                                      |
+| out float **NORMALMAP_DEPTH**    | Depth from variable above. Defaults to 1.0.                                                      |
 +----------------------------------+--------------------------------------------------------------------------------------------------+
 | in vec2 **UV**                   | UV that comes from vertex frunction.                                                             |
```
Comment (Noshyaar) [inline learning/features/shading/shading_language.rst]: `frunction`

**Comment by Noshyaar** [comment]
Line 12: implementaiton
Line 129: whathever
Line 154: a newline is needed between codeblock header and content
Line 885: Substractive
Line 956: Missing one `^`

**Comment by eluadyl** [comment]
Thanks for helping. Made those changes, and some more I caught too. I'll read over once more and commit again.

**Comment by NathanLovato** [comment]
Seems good for now. We'd need some people to do some edit work - rephrase and simplify the docs, partly so they conform to the writing guidelines, and because it'll become shorter and be easier to read.

**Pull request merged**